shared/actions/fs/index.js

```diff
@@ -690,96 +691,108 @@ const closeDestinationPicker = (state, action) => {
+      oneChatPerTLF: false,
+      tlfName: elems[2],
+      topicName: '',
+      topicType: RPCChatTypes.commonTopicType.chat,
+      visibility: RPCTypes.commonTLFVisibility.private,
+    }).then(result =>
+      // This action, no matter setting a real idKey or
+      // noConversationIDKey, causes a transition into 'read-to-send'
+      // state, which is what we want here. If we don't have a
+      // conversation we should create it when user tries to send.
+      FsGen.createSetSendLinkToChatConvID({
+        convID:
+          result.conversations && result.conversations.length
+            ? ChatTypes.conversationIDToKey(result.conversations[0].info.id)
+            : ChatConstants.noConversationIDKey,
       })
     )
-  } else {
-    // It's a real team, but we don't know if it's a small team or big team. So
-    // call RPCChatTypes.localGetTLFConversationsLocalRpcPromise to get all
-    // channels. We could have used the Teams store, but then we are doing
-    // cross-store stuff and are depending on the Teams store. If this turns
-    // out to feel slow, we can probably cahce the results.
-    actions.push(
-      Saga.callUntyped(function*() {
-        const result = yield Saga.callPromise(RPCChatTypes.localGetTLFConversationsLocalRpcPromise, {
-          membersType: RPCChatTypes.commonConversationMembersType.team,
-          tlfName: elems[2],
-          topicType: RPCChatTypes.commonTopicType.chat,
-        })
-
-        if (!result.convs || !result.convs.length) {
-          // TODO: error?
-          return
-        }
+  }
 
-        yield Saga.put(
+  // It's a real team, but we don't know if it's a small team or big team. So
+  // call RPCChatTypes.localGetTLFConversationsLocalRpcPromise to get all
+  // channels. We could have used the Teams store, but then we are doing
+  // cross-store stuff and are depending on the Teams store. If this turns
+  // out to feel slow, we can probably cahce the results.
+
+  return RPCChatTypes.localGetTLFConversationsLocalRpcPromise({
+    membersType: RPCChatTypes.commonConversationMembersType.team,
+    tlfName: elems[2],
+    topicType: RPCChatTypes.commonTopicType.chat,
+  }).then(result =>
+    !result.convs || !result.convs.length
+      ? null // TODO: is this possible for teams at all?
+      : [
           FsGen.createSetSendLinkToChatChannels({
-            channels: I.Map(result.convs.map(conv => [conv.convID, conv.channel])),
-          })
-        )
+            channels: I.Map(
+              result.convs.map(conv => [ChatTypes.stringToConversationIDKey(conv.convID), conv.channel])
+            ),
+          }),
+
+          ...(result.convs && result.convs.length === 1
+            ? [
+                // Auto-select channel if it's the only one.
+                FsGen.createSetSendLinkToChatConvID({
+                  convID: ChatTypes.stringToConversationIDKey(result.convs[0].convID),
+                }),
+              ]
+            : []),
+        ]
```

Comment:
Shouldn't this return a promise?

Reply:
This is used in `chainAction`, so the resolved result is treated as an action or an array of actions to be put into the store. It _can_ return a promise, which would just continue resolving until it's not a promise anymore. In this case, these actions are the final result.